boxes/boxes/vanilla/app/embedded-wallet.ts

```diff
@@ -81,19 +82,24 @@ export class EmbeddedWallet extends BaseWallet {
     const maxFeesPerGas =
       userFeeOptions?.gasSettings?.maxFeesPerGas ??
       (await this.aztecNode.getCurrentBaseFees()).mul(1 + this.baseFeePadding);
-    const sponsoredFPCContract =
-      await EmbeddedWallet.#getSponsoredPFCContract();
-    let paymentMethod;
-    let endSetup = false;
-    let isFeePayer = false;
+    let walletFeePaymentMethod;
+    let accountFeePaymentMethodOptions;
     // The transaction does not include a fee payment method, so we set a default
     if (!userFeeOptions?.embeddedPaymentMethodFeePayer) {
-      paymentMethod = new SponsoredFeePaymentMethod(
+      const sponsoredFPCContract =
+        await EmbeddedWallet.#getSponsoredPFCContract();
+      walletFeePaymentMethod = new SponsoredFeePaymentMethod(
         sponsoredFPCContract.instance.address
       );
+      accountFeePaymentMethodOptions = AccountFeePaymentMethodOptions.EXTERNAL;
```

Comment:
Would `walletFeePaymentMethod` ever be set of the option is not external?

Reply:
It should always be set to *something*...but just in case, I shuffled the options around and assigned the "EXTERNAL" option to value 0 in the enum. It lines up better with "do nothing unless explicitly stated"

Reply:
Yes, if it's set to "fee juice with claim"